pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -646,6 +646,14 @@ func (c *Controller) syncRule(key string) error {
 		return nil
 	}
 
+	if !c.antreaProxyEnabled && rule.isToServicesPolicyRule() {
```

Comment:
Should we handle both cases (applied to service / selects services as peers with toServices) separately?
It would make sense to me because:
1) the first case (appliedTo) doesn't only require AntreaProxy, I think it also requires proxyAll? Should we add this to the validation?
2) we may be able to generate a better error message
3) `isToServicesPolicyRule` is not a great function name IMO; the name implies that we only consider the second case (`toServices`)

Reply:
> 1. the first case (appliedTo) doesn't only require AntreaProxy, I think it also requires proxyAll? Should we add this to the validation?

The feature `applied to Service` should only require AntreaProxy. It is added in this #2755.  Do you mean that we should add validation here https://github.com/antrea-io/antrea/blob/9e84a682e0b522dffdc8fcf7cb314ab31eb18903/pkg/agent/controller/networkpolicy/networkpolicy_controller.go#L621? For the first case, the expected result of `realizable` should be false. But it is a little strange that adding `antreaProxyEnabled` member to struct `ruleCache`.

> 2. we may be able to generate a better error message

Do you mean that we should generate more detailed error log here?

> 3 `isToServicesPolicyRule` is not a great function name IMO; the name implies that we only consider the second case (toServices)

How about `isAppliedToServiceIngressPolicyRule()`  and `isToServicesEgressPolicyRule()`?

Reply:
I think the `appliedTo` case (different from `toServices` as an egress rule) was added in https://github.com/antrea-io/antrea/pull/3997, and does require ProxyAll?
@tnqn am I missing something?

Reply:
@antoninbas Sorry for updating late. I have verified it with @GraysonWu. It does require `proxyAll` for `appliedTo` case.